brave-intl/publishers
# Publisher Status Updates API 🏷

## Publisher Status Updates API 🏷 

#### API
---
`GET` to `api/v1/publishers/14731e0b-a795-47d3-b2c9-b30d06df720f`

```ruby
{
  "id": "14731e0b-a795-47d3-b2c9-b30d06df720f",
  "owner_identifier": "publishers#uuid:14731e0b-a795-47d3-b2c9-b30d06df720f",
  "email": "user@test.com",
  "name": "Test",
  "created_at": "2019-04-20T13:08:43.382-07:00",
  "current_status": {
    "status": "active",
    "created_at": "2019-04-22T14:55:55.276-07:00"
  }
}
```
---

`POST` to `api/v1/publishers/14731e0b-a795-47d3-b2c9-b30d06df720f/publisher_status_updates`

Example body
```ruby
{
  "status": "Suspended",
  "note": "Justification for suspension - link-to-evidence.com/12345",
  "admin": "admin@brave.com"
}
```

Response
```ruby
{
  "success": "true"
}
```
---

#### Error Handling

``` ruby
{
  "error": "Not Found",
  "detail": "Publisher with id xyz-a795-47d3-b2c9-b30d06df72f0f not found"
}
```

```ruby 
{
  "error": "Status Invalid",
  "detail": "Status abc is not valid, please use one of the following: created, onboarding, active, suspended, locked"
}
```

```ruby
{
  "error": "Admin Invalid",
  "detail": "Admin field cannot be null, please provide e-mail of an admin"
}
```

```ruby
{
  "error": "Note Invalid",
  "detail": "Note cannot be null, please provide justification for status update"
}
```

#### Features

* API for updating of publisher status
* `POST` request requires a valid state, note for justification, and e-mail of a publisher admin account
* Closes #1800 

#### How To Test

1. Create a new publisher
2. From a REST client try `GET` and `POST` Requests to the api above 
3. As an admin, view the publisher and check the status updates match the api calls. 

#### Pull Request Checklist
